activesupport/lib/active_support/dependencies.rb

```diff
@@ -665,7 +665,7 @@ def new_constants_in(*descs)
         new_constants = constant_watch_stack.new_constants
 
         return new_constants unless aborting
-
+        # Removing partially loaded constants when error occurred during loading.
```

Comment:
There's something strange here, already present before we removed the traces: the comment says an error condition is going on if you reached this point. This should be obvious from reading the source.

I think this could be rewritten to be more clear. Instead of `aborting` let's use a flag with the opposite meaning like `success`. Then we can just say:

```ruby
if success
  new_constants
else
  # Remove partially loaded constants.
  new_constants.each { |c| remove_constant(c) }.clear
end
```

The intention to reverse the flag is to have the optimistic path first in the if/else.